### PR TITLE
Directly use WebSMS API, send ID

### PR DESCRIPTION
**Pull request by adambrown1**
Directly use the websms API instead of email to SMS conversion. Gives us more control. Importantly we can set the message sender number. This is important so people can add it to their contacts / bypass do not disturb / set a loud ringtone for that contact etc.

There is probably a clever way to combine send_sms and send_phonecall, and allow for flexibility for future services. In the end they are all post requests that take some set of parameters stored in the database and send the message with a specific key, getting back a response code. I started to go in this direction with the generic "postparameters" database entry, but more work is needed (variable message formatting, response code interpretation) and there are more pressing things

### CODE REVIEW

File: Doberman/AlarmMonitor.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def send_phonecall(self, phone_numbers, message):
             self.logger.warning(f"Message exceeds {maxmessagelength} "
                                 "characters. Message will be shortened.")
         message = f"This is the {self.db.experiment_name} alarm system. " + message
-        if len(phone_numbers) == 1:
+        if type(phone_numbers) == str:
```
Comment (darrylmasson) [inline Doberman/AlarmMonitor.py:58]: I would do `isinstance(phone_numbers, str)` here rather than `type`

File: Doberman/AlarmMonitor.py [hunk 2]
```diff
@@ -121,47 +121,43 @@ def send_email(self, toaddr, subject, message, cc=None, bcc=None, add_signature=
...
+            raise ValueError("No phone number given.")
+
+        message = str(message)
+        # Long messages are shortened to avoid excessive fees
+        if len(message) > maxmessagelength:
+            message = ' '.join(message[:maxmessagelength + 1].split(' ')[0:-1])
+            self.logger.warning(f"Message exceeds {maxmessagelength} "
+                                "characters. Message will be shortened.")
+
+        if type(phone_numbers) == str:
```
Comment (darrylmasson) [inline Doberman/AlarmMonitor.py:148]: Again here

File: Doberman/AlarmMonitor.py [hunk 3]
```diff
@@ -121,47 +121,43 @@ def send_email(self, toaddr, subject, message, cc=None, bcc=None, add_signature=
...
+            phone_numbers = [phone_numbers]
+        for tonumber in phone_numbers:
+            data = postparameters
+            data['Recipient'] = tonumber
+            data['SMSText'] = message
+            self.logger.warning(f'Sending SMS to {tonumber}')
+            print(data)
+            response = requests.post(url, data=data)
+            print(response.status_code)
```
Comment (darrylmasson) [inline Doberman/AlarmMonitor.py:157]: These two print statements should either get removed or changed to log calls

File: Doberman/AlarmMonitor.py [hunk 4]
```diff
@@ -121,47 +121,43 @@ def send_email(self, toaddr, subject, message, cc=None, bcc=None, add_signature=
...
-                               subject=subject,
-                               message=message,
-                               cc=cc, bcc=bcc,
-                               add_signature=False) == -1:
-                self.logger.error('Could not send SMS! Email to SMS not working.')
-                return -1
+        url = connection_details['url']
+        postparameters = connection_details['postparameters']
+        maxmessagelength = int(connection_details['maxmessagelength'])
+        auth = connection_details.get('auth')
```
Comment (darrylmasson) [inline Doberman/AlarmMonitor.py:137]: Is this used?

**Review by darrylmasson — CHANGES_REQUESTED** [review]
Overall looks fine, a couple of minor tweaks are recommended

File: Doberman/AlarmMonitor.py (hunk 4 above)
Comment (adambrown1) [reply Doberman/AlarmMonitor.py]: Removed

File: Doberman/AlarmMonitor.py (hunk 3 above)
Comment (adambrown1) [reply Doberman/AlarmMonitor.py]: They're gone

**Review by darrylmasson — APPROVED** [review]

**Pull request merged**